### PR TITLE
refactored payment of reservation with stellar

### PR DESCRIPTION
**Pull request by DylanVerstraete**
Work in progress

### CODE REVIEW

File: JumpscaleLibs/sal/zosv2/billing.py [hunk 1]
```diff
@@ -1,59 +1,33 @@
...
+        else:
+            asset = ASSET_CODE + ":" + TFT_ISSUER_PROD
+
+        reservation_id = reservation_response["id"]
+        escrow_informations = reservation_response["escrow_information"]
+        for escrow in escrow_informations:
+            escrow_address = escrow["escrow_address"]
+            total_amount = escrow["total_amount"] / 10e6
+            try:
+                client.transfer(escrow_address, total_amount, asset=asset, memo_text=str(reservation_id))
```
Comment (zaibon) [inline JumpscaleLibs/sal/zosv2/billing.py]: shouldn't this methods return the list of transaction if that has been created or something ?
So user can followup on these transaction if he wants to ?

File: JumpscaleLibs/sal/zosv2/billing.py [hunk 2]
```diff
@@ -1,59 +1,33 @@
...
+            except BadRequestError as e:
+                self._log_debug(e)
```
Comment (zaibon) [inline JumpscaleLibs/sal/zosv2/billing.py]: What does this exception means? I do not like we mask errors like this, better to just let it bubble up

File: JumpscaleLibs/sal/zosv2/readme.md [hunk 3]
```diff
@@ -259,6 +257,40 @@ zos.volume.attach_existing(
...
+
+# if you don't have a trustline to the issuer of TFT on stellar
+# trustline for testnet
+client.add_trustline("TFT", "GA47YZA3PKFUZMPLQ3B5F2E3CJIB57TGGU7SPCQT2WAEYKN766PWIMB3")
+
+# trustline for production
+client.add_trustline("TFT", "GBOVQKJYHXRR3DX6NOX2RRYFRCUMSADGDESTDNBDS6CDVLGVESRTAC47")
+
+zos = j.sal.zosv2
+zos._explorer = explorer
```
Comment (zaibon) [inline JumpscaleLibs/sal/zosv2/readme.md]:
```suggestion
```

Comment (zaibon) [reply JumpscaleLibs/sal/zosv2/readme.md]: This is not needed, the explorer address should already properly configured

**Review by zaibon — DISMISSED** [review]

**Review by zaibon — CHANGES_REQUESTED** [review]
did a first round of review, couple of comments

**Review by xmonader — CHANGES_REQUESTED** [review]
looks good

- reservation_register is specified to return an int 
https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/sal/zosv2/ZOSv2.py#L68
while it actually returns json https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/clients/explorer/reservations.py#L19 and you're treating it as json already

- please update the reservation model to have the escrow_details defined as well

**Comment by DylanVerstraete** [comment]
> looks good
> 
> * reservation_register is specified to return an int
>   https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/sal/zosv2/ZOSv2.py#L68
>   while it actually returns json https://github.com/threefoldtech/jumpscaleX_libs/blob/master/JumpscaleLibs/clients/explorer/reservations.py#L19 and you're treating it as json already
> * please update the reservation model to have the escrow_details defined as well


Thanks ! 

Added a new model to the threebot:

https://github.com/threefoldtech/jumpscaleX_threebot/pull/509

**Review by zaibon — CHANGES_REQUESTED** [review]
readme still need to be update to use the new `reservation_create` model

**Review by zaibon — APPROVED** [review]

**Pull request merged**